### PR TITLE
UCP/PROTO/MULTI: Sort lanes by BW

### PR DESCRIPTION
**Pull request by iyastreb**
## What
While working on [RM#4134043](https://redmine.mellanox.com/issues/4134043), we discovered that lanes for multi-frag protocols are not sorted by bandwidth. It's desirable to sort them in order to use lanes with higher bandwidth first

## Why ?
As discussed with UCX team, in multi-frag protocols it's desirable to have lanes sorted by bandwidth in order to optimize sending. First we try to send message over lane with highest bandwidth, and therefore potentially avoid splitting a message into several parts.

## How ?
Sort lanes by BW after picking them and detecting BW

### CODE REVIEW

File: src/ucp/proto/proto_multi.c [hunk 1]
```diff
@@ -86,6 +99,15 @@ ucs_status_t ucp_proto_multi_init(const ucp_proto_multi_init_params_t *params,
         max_bandwidth = ucs_max(max_bandwidth, lane_perf->bandwidth);
     }
 
+    /* Sort lanes by bandwidth */
+    if (num_lanes > 1) {
+        /* If type of the first lane is not the same as in the middle, then
+         * exclude this lane from sorting */
```
Comment (brminich) [inline src/ucp/proto/proto_multi.c:105]: this comments just repeats what the code does. Can you pls descrive the reason for that? E.g. that when the first lane is different is should always be used first e.g. to preserve ordering with eager protocols for TAG and AM APIs.

File: src/ucp/proto/proto_multi.c [hunk 2]
```diff
@@ -86,6 +99,15 @@ ucs_status_t ucp_proto_multi_init(const ucp_proto_multi_init_params_t *params,
...
+        lane = (params->first.lane_type == params->middle.lane_type) ? 0 : 1;
```
[the remark this answers is not in the document]
Comment (iyastreb) [reply src/ucp/proto/proto_multi.c:106]: sure

File: src/ucp/proto/proto_multi.c (hunk 1 above)
Comment (iyastreb) [reply src/ucp/proto/proto_multi.c:105]: ok

**Review by yosefe — APPROVED** [review]

**Review by brminich — APPROVED** [review]

**Pull request merged**